File: Data Structures & Algorithms/anagram-groups/submission-1.js
class Solution {
    /**
     * @param {string[]} strs
     * @return {string[][]}
     */
    groupAnagrams(strs) {

        const map = new Map()
        for(let i=0;i<strs.length;i++){
            const hash = this.getStringHash(strs[i])
            let temp =[]
            if(map.has(hash)){
                temp = map.get(hash)
            }
            temp.push(strs[i])
            map.set(hash, temp)
        }
        const result = []
        map.forEach((value, key) => {
            result.push(value)
        })
        return result;
    }

    getStringHash (s){
        const letters = Array(26).fill(0);
        for(let i =0;i<s.length;i++){
            letters[s.charCodeAt(i) - 97]++
        }
        return letters.join(',')
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {

        int i=0;
        int j=numbers.length -1;
        int []result = new int[2];
        while(i < j){
            if(numbers[j] > (target - numbers[i])){
                j--;
            }
            else if(numbers[j] < (target - numbers[i])){
                i++;
            }
            else{
                result[0] = i + 1;
                result[1] = j + 1;
                break;
            }
        }
        return result;
        
    }
}
